doc/reference/bluetooth/mesh/shell.rst

```diff
@@ -0,0 +1,730 @@
+
+	uart:~$ mesh provision 0 0x0001
+
+Since all mesh shell nodes use the same values for the default network key, this can be done on multiple devices, as long as they're assigned non-overlapping unicast addresses. Alternatively, to provision the device into an existing network, the unprovisioned beacon can be enabled with ``mesh pb-adv on`` or ``mesh pb-gatt on``. The beacons can be picked up by an external provisioner, which can provision the node into its network.
+
+Once the mesh node is part of a network, its transmission parameters can be controlled by the general configuration commands:
+
+* To set the destination address, call ``mesh dst <addr>``.
+* To set the network key index, call ``mesh netidx <NetIdx>``.
+* To set the application key index, call ``mesh appidx <AppIdx>``.
+
+By default, the transmission parameters are set to send messages to the provisioned address and network key.
+
+Configuration
+=============
+
+By setting the destination address to the local unicast address (``0x0001`` in the ``mesh provision`` command above), we can perform self-configuration through any of the :ref:`bluetooth_mesh_shell_cfg_cli` commands.
+
+A good first step is to read out the node's own composition data::
+
+	uart:~$ mesh get-comp
+
+This prints a list of the composition data of the node, including a list of its model IDs.
+
+Next, since the device has no application keys by default, it's a good idea to add one::
+
+	uart:~$ mesh app-key-add 0 0
+
+Message sending
+===============
+
+With an application key added (see above), the mesh node's transition parameters are all valid, and the Mesh Shell can send raw mesh messages through the network.
+
+For example, to send a Generic OnOff Set message, call::
+
+	uart:~$ mesh net-send 82020100
+
+The message will be sent to the current destination address, using the current network and application key indexes. As the destination address points to the local unicast address by default, the device will only send packets to itself. To change the destination address to the All Nodes broadcast address, call::
+
+	uart:~$ mesh dst 0xffff
+
+With the destination address set to ``0xffff``, any other mesh nodes in the network with the configured network and application keys will receive and process the messages we send.
+
+.. note::
+	To change the configuration of the device, the destination address must be set back to the local unicast address before issuing any configuration commands.
+
+Sending raw mesh packets is a good way to test model message handler implementations during development, as it can be done without having to implement the sending model. By default, only the reception of the model messages can be tested this way, as the Mesh Shell only includes the foundation models. To receive a packet in the Mesh Shell node, you have to add a model with a valid opcode handler list to the composition data, and print the incoming message to the shell in the handler callback.
```

Comment:
" To receive a packet in the Mesh Shell node, you have to add a model with a valid opcode handler list to the composition data, and print the incoming message to the shell in the handler callback."
This has to be done in code? No shell commands to do this, right?

Reply:
Yes, good point. Added "in shell.c"